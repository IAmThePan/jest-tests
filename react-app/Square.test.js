import React from 'react';
import { Square } from './cmpts';
import renderer from 'react-test-renderer';

test('Snapshot Tests for Square', () => {
  const component = renderer.create(<Square onClick={() => {}} value="" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const componentX = renderer.create(<Square onClick={() => {}} value="X" />);
  const treeX = componentX.toJSON();
  expect(treeX).toMatchSnapshot();

  const componentO = renderer.create(<Square onClick={() => {}} value="O" />);
  const treeO = componentO.toJSON();
  expect(treeO).toMatchSnapshot();
});
