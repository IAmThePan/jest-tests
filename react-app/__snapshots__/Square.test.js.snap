// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests for Square 1`] = `
<button
  className="square"
  onClick={[Function]}
>
  
</button>
`;

exports[`Snapshot Tests for Square 2`] = `
<button
  className="square"
  onClick={[Function]}
>
  X
</button>
`;

exports[`Snapshot Tests for Square 3`] = `
<button
  className="square"
  onClick={[Function]}
>
  O
</button>
`;
